nums/core/linalg.py

```diff
@@ -256,6 +256,160 @@ def inv(app: ArrayApplication, X: BlockArray):
+                Z_oid = Zs.blocks[(row_block, col_block)].oid
+
+                # Calculate -R00 = 0 - R00
+                neg_R00_oid = app.cm.bop(
+                    "subtract",
+                    Z_oid,
+                    R00_oid,
+                    False,
+                    False,
+                    axes=1,
+                    syskwargs={
+                        "grid_entry": (row_block, col_block),
+                        "grid_shape": grid_shape,
+                    },
+                )
+                # Calculate -R00 @ R01
+                sub_oids.append(
+                    app.cm.bop(
+                        "tensordot",
+                        neg_R00_oid,
+                        R01_oids[col_block],
+                        False,
+                        False,
+                        axes=1,
+                        syskwargs={
+                            "grid_entry": R01_grid_entries[col_block],
+                            "grid_shape": grid_shape,
+                        },
+                    )
+                )
+
+            # Finished with one blocked mult
+            R01_1_oids.append(
+                app.cm.sum_reduce(
+                    *sub_oids,
+                    syskwargs={
+                        "grid_entry": R01_grid_entries[row_block],
+                        "grid_shape": grid_shape,
+                    }
+                )
+            )
+
+        # Perform matrix multiplication: R_01_2 = R_01_1 @ R_11_inv
+        R01_2_oids = []
+        for row_block in range(R01_num_blocks):
+            R01_2_oids.append(
+                app.cm.bop(
+                    "tensordot",
+                    R01_1_oids[row_block],
+                    R11_inv_oid,
+                    False,
+                    False,
+                    axes=1,
+                    syskwargs={
+                        "grid_entry": R01_grid_entries[row_block],
+                        "grid_shape": grid_shape,
+                    },
+                )
+            )
+
+        # Reset R_01
+        for i, entry in enumerate(R01_grid_entries):
+            R.blocks[entry].oid = R01_2_oids[i]
+
+        # Recompute R_tl.shape
+        r11_r, r11_c = R11_shape
+        old_r, old_c = R_tl_shape
+
+        R_tl_shape = (old_r + r11_r, old_c + r11_c)
+
+    # By the time we finish, R = R_inv
+    return R
+
+
+def inv_cholesky(app: ArrayApplication, X: BlockArray):
+    # Matrix Inversion for X where X is a square positive definite matrix
+    # Particularly useful for least-squares regression on a large data corpus
+
+    # Step 1: Calculate the U using Cholesky, where X = U^TU
+    U = app.cholesky_blocked(X)
+
+    # Step 2: Compute U^-1
+    U_inv = app.inv_uppertri(U)
+
+    # Step 3: Compute inv(X) by U_inv @ U_inv.T
+    return U_inv @ U_inv.T
+
+
 def cholesky(app: ArrayApplication, X: BlockArray):
     # TODO (hme): Implement scalable version.
```

Comment:
in my understanding, the ```cholesky_blocked``` is the scalable version for this? @elibol

Reply:
I think so, so long as the assumptions made about the input matrix to `choleskly_blocked` are the same as the NumPy implementation.

Reply:
Is it okay if I keep it in this PR? May be a useful implementation down the line.

Reply:
yes, looks good